# Objetivo: reportar datos faltantes en un DataFrame de pandas

In [11]:
import pandas as pd

In [12]:
df = pd.read_csv('empleados.csv', delimiter=";")

In [13]:
df.head()

,Nombre,Apellido,Sexo,Estado_Civil,Edad,Experiencia,Salario,Gerencia,Departamento
0,Alexandro,NaN,Masculino,Soltero(a),22.0,3.0,2423.0,No,Publicidad
1,Carlos,NaN,Masculino,Casado(a),38.0,10.0,3132.0,Sí,NaN
2,Felipa,NaN,Femenino,Soltero(a),26.0,5.0,2639.0,No,Comercial
3,Daniel,NaN,Masculino,Casado(a),35.0,10.0,3122.0,No,Comercial
4,John,NaN,Masculino,Casado(a),35.0,13.0,3431.0,Sí,Servicios al cliente


In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 251 entries, 0 to 250
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Nombre        245 non-null    object 
 1   Apellido      0 non-null      float64
 2   Sexo          214 non-null    object 
 3   Estado_Civil  247 non-null    object 
 4   Edad          190 non-null    float64
 5   Experiencia   247 non-null    float64
 6   Salario       245 non-null    float64
 7   Gerencia      230 non-null    object 
 8   Departamento  239 non-null    object 
dtypes: float64(4), object(5)
memory usage: 17.8+ KB


Se puede observar que el DataFrame contiene 9 columnas distintas con 251 entradas en total. A partir de la función `info()` se puede observar que todas las columnas tienen algunos datos faltantes. En el otro extremo, la columna de `Apellido` no cuenta con ningún dato, por lo que una revisión de la causa sería precisa 

In [15]:
# Eliminación de la columna sin datos
df.drop(columns=['Apellido'], inplace= True)

In [16]:
# Conteo de NaN por columna
nan_counts = df.isna().sum()
print("Conteo de NaN por columna: ")
print(nan_counts)

Conteo de NaN por columna: 
Nombre           6
Sexo            37
Estado_Civil     4
Edad            61
Experiencia      4
Salario          6
Gerencia        21
Departamento    12
dtype: int64


In [17]:
# Eliminación de filas con datos faltantes
df.dropna(inplace= True)

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 145 entries, 0 to 250
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Nombre        145 non-null    object 
 1   Sexo          145 non-null    object 
 2   Estado_Civil  145 non-null    object 
 3   Edad          145 non-null    float64
 4   Experiencia   145 non-null    float64
 5   Salario       145 non-null    float64
 6   Gerencia      145 non-null    object 
 7   Departamento  145 non-null    object 
dtypes: float64(3), object(5)
memory usage: 10.2+ KB


Actualmente todas las filas con algún dato faltante que es automáticamente reconocido por pandas como `NaN`han sido eliminadas del DataFrame, sin embargo, pueden seguir existiendo valores erróneos o algún tipo de dato faltante que no haya sido reconocido como tal. Por ende se realizará una revisión más exhaustiva sobre la naturaleza de los datos que quedan

In [19]:
df.columns

Index(['Nombre', 'Sexo', 'Estado_Civil', 'Edad', 'Experiencia', 'Salario',
       'Gerencia', 'Departamento'],
      dtype='object')

In [24]:
for i in df.columns:
  print(f'Valores únicos en la columna "{i}": \n{df[i].unique()}\n')

Valores únicos en la columna "Nombre": 
['Alexandro' 'Felipa' 'Daniel' 'John' 'Ruperta' 'Francisca' 'Selena'
 'Ximena' 'David' 'Max' 'Paula' 'Tim' 'Gloria' 'Jesse' 'Ronald' 'Manola'
 'Isabela' 'Herberto' 'Lilia' 'Agustina' 'Anselma' 'Lucho' 'Maristela'
 'Gabriela' 'Olegario' 'Berta' 'Albino' 'Albert' 'Florina' 'Soledad'
 'Miriam' 'Lucio' 'Luisa' 'Regina' 'Dionisio' 'Eliseo' 'Loreto' 'Olivia'
 'Alfonso' 'Carlota' 'Eligia' 'Amada' 'Amado' 'Cirino' 'Joaquina'
 'Susanita' 'Melisa' 'Yolanda' 'Paulino' 'Teodora' 'Herminia' 'Francisco'
 'Pastora' 'Jose' 'Eva' 'Erasmo' 'Eliana' 'Adalberto' 'Ana' 'Aureliano'
 'Samanta' 'Carmelo' 'Consuelo' 'Teo' 'Leandro' 'Fanny' 'Dionisia' 'Leire'
 'Lorenzo' 'Hilario' 'Flora' 'Celso' 'Luz' 'Adelia' 'Santiago' 'Fulgencio'
 'Clarisa' 'Clara' 'Soraya' 'Reyes' 'Iker' 'Ernesto' 'Estela' 'Amancio'
 'Lino' 'Juan' 'Adela' 'Eugenio' 'Daisy' 'Marta' 'Elodia' 'Emilio' 'Kevin'
 'Belen' 'Richard' 'Aaron' 'Douglas' 'Sean' 'Celia' 'Angela' 'Olga'
 'Rafaela' 'Corey' 'Otilia' 